src/main/java/com/box/sdk/BoxFile.java

```diff
@@ -1038,6 +1038,33 @@ public Metadata createMetadata(String typeName, String scope, Metadata metadata)
         return new Metadata(JsonObject.readFrom(response.getJSON()));
     }
 
+    /**
+     * Attempts to create a metadata value on a file. If conflict then perform an update of metadata values on file.
+     *
+     * @param templateName the name of the metadata template.
+     * @param scope        the scope of the template (usually "global" or "enterprise").
+     * @param metadata     the new metadata values.
+     * @return the metadata returned from the server.
+     */
+    public Metadata setMetadata(String templateName, String scope, Metadata metadata) {
+        Metadata metadataValue = null;
+
+        try {
+            metadataValue = this.createMetadata(templateName, scope, metadata);
+        } catch (BoxAPIException e) {
+            if (e.getResponseCode() == 409) {
+                Metadata metadataToUpdate = new Metadata(scope, templateName);
+                for (JsonValue value : metadata.getOperations()) {
+                    metadataToUpdate.add(value.asObject().get("path").asString(),
+                            value.asObject().get("value").toString());
```

Comment:
This line worries me — does this work with all the data types we currently use for metadata?  In your tests, could you verify that this works correctly with numbers and arrays of strings (for `multiSelect` metadata)?